### ISSUE TITLE
Animation not picking up the current set/animated value

### ISSUE THREAD

**Issue by eriklimakc** [unoplatform/uno#14262]
### Current behavior
This happens in all platforms except Windows.

![Skia_vs_Win_issue](https://github.com/unoplatform/Uno.Themes/assets/114886335/ed545760-6386-4faf-aa84-f49b2e77a3b4)

Apparently when the animation code is executed, the Text, Header, and Placeholder are quickly set to their default state and after that they are animated.

### How to reproduce it (as minimally and precisely as possible)
Checkout [this branch](https://github.com/unoplatform/Uno.Themes/tree/dev/ERLI/textbox-animation-repro) on Uno.Themes and launch on non-windows platform.

For the FilledTextBoxStyle the animation is defined in: [TextBox.xaml#L420](https://github.com/unoplatform/Uno.Themes/blob/dev/ERLI/textbox-animation-repro/src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml#L420).

### Workaround

In order to test I replaced the animation for `Setters` like below and we don't have that behavior.

```
<Setter Target="HeaderElement_CompositeTransform.TranslateY"
	Value="-11" />
<Setter Target="ContentElement_CompositeTransform.TranslateY"
	Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource EmptyToFocusedContentCompositeTransformTranslateY}, TargetNullValue=0, FallbackValue=0}" />
<Setter Target="PlaceholderElement_CompositeTransform.TranslateY"
	Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource EmptyToFocusedContentCompositeTransformTranslateY}, TargetNullValue=0, FallbackValue=0}" />
<Setter Target="HeaderElement_CompositeTransform.ScaleX"
	Value="0.7" />
<Setter Target="HeaderElement_CompositeTransform.ScaleY"
	Value="0.7" />
```

### Environment
Works on UWP/WinUI: YES
NuGet package version(s): Uno.WinUI@5.0.0-dev.2572
Affected platforms: This happens in all platforms except Windows.

### Anything else we need to know?
Comment left by @Xiaoy312:

_"At a glance, this seems like an issue with the animation not picking up the current set/animated value as the starting point for the (start...end) animation. This is not probably not something we will fix here, but in uno."_

**Comment by agneszitte** [comment]
cc @MartinZikmund / @dr1rrb / @Xiaoy312

**Comment by agneszitte** [comment]
@jeromelaban / @MartinZikmund this issue kind of blocking @eriklimakc's PR here https://github.com/unoplatform/Uno.Themes/pull/1014
(cc @kazo0, @Xiaoy312)

**Comment by kazo0** [comment]
Since this will cause a regression in the animation for all Uno platforms, I'd say it's definitely blocking us from merging https://github.com/unoplatform/Uno.Themes/pull/1014

**Comment by kazo0** [comment]
@dr1rrb does this happen to look familiar at all? Currently blocking this https://github.com/unoplatform/Uno.Themes/issues/974

**Comment by dr1rrb** [comment]
No nothing familiar here, sorry :(

**Comment by Xiaoy312** [comment]
looks like a regression from #11859 ?

**Comment by agneszitte** [comment]
@eriklimakc I imagine this is still happening with latest versions?

**Comment by eriklimakc** [comment]
Apparently fixed on latest Uno packages. Quickly tested all platforms and didn't see the issue anymore. You can also see on the stage site: https://salmon-rock-0cfebe70f-1014.eastus2.azurestaticapps.net/ from https://github.com/unoplatform/Uno.Themes/pull/1014.

Will do more deep tests and update here on the issue.

cc @Xiaoy312 @agneszitte @kazo0

**Comment by Xiaoy312** [comment]
unassigned me and close this, if your deep tests are positive.

**Comment by eriklimakc** [comment]
Closing as not possible to reproducible anymore.

**Comment by eriklimakc** [comment]
@kazo0 It was fixed on [5.3.0-dev.906](https://www.nuget.org/packages/Uno.WinUI/5.3.0-dev.906).

**Comment by Youssef1313** [comment]
For reference, that version belongs to https://github.com/unoplatform/uno/pull/16474

FYI @ramezgerges :)
